backend/sirius-components-diagrams-layout/src/main/java/org/eclipse/sirius/components/diagrams/layout/incremental/IncrementalLayoutEngine.java

```diff
@@ -181,18 +207,32 @@ private void layoutBorderNodes(Optional<IDiagramEvent> optionalDiagramElementEve
             }
 
             // 2- recompute the border node
-            EnumMap<Geometry.RectangleSide, List<NodeLayoutData>> borderNodesPerSide = this.snapBorderNodes(borderNodesLayoutData, initialNodeBounds.getSize(), layoutConfigurator);
+            List<BorderNodesOnSide> borderNodesPerSide = this.snapBorderNodes(borderNodesLayoutData, initialNodeBounds.getSize(), layoutConfigurator);
 
             // 3 - move the border node along the side according to the side change
             this.updateBorderNodeAccordingParentResize(optionalDiagramElementEvent, initialNodeBounds, newNodeBounds, borderNodesPerSide, borderNodesLayoutData.get(0).getParent().getId());
+
+            // 4- set the label position if the border is newly created
+            this.updateBorderNodeLabel(optionalDiagramElementEvent, borderNodesPerSide);
+        }
+    }
+
+    private void updateBorderNodeLabel(Optional<IDiagramEvent> optionalDiagramElementEvent, List<BorderNodesOnSide> borderNodesPerSideList) {
+
```

Comment:
Why all those empty lines while starting some method (that's not the first time)?

Reply:
done